dartagnan/src/main/java/com/dat3m/dartagnan/program/event/EventFactory.java

```diff
@@ -602,4 +604,44 @@ public static Fence newLwSyncBarrier() {
         }
     }
 
-}
+    // =============================================================================================
+    // ============================================ PTX ============================================
+    // =============================================================================================
+    public static class PTX {
+        private PTX() {}
+
+        public static Store newTaggedStore(IExpr address, ExprInterface value, String mo,  String scope) {
+            Store store = new Store(address, value, mo); // mo = REL || RLX || WEAK
+            store.addFilters(scope); // scope = CTA || GPU || SYS
+            return store;
+        }
+
+        public static Load newTaggedLoad(Register register, IExpr address, String mo, String scope) {
+            Load load = new Load(register, address, mo); // mo = ACQ || RLX || WEAK
+            load.addFilters(scope); // scope =  CTA || GPU || SYS
+            return load;
+        }
+
+        public static Fence newTaggedFence(String mo, String scope) {
+            Fence fence = new Fence(mo); // mo = ACQ_REL || SC
+            fence.addFilters(scope);
+            return fence;
+        }
+
+        public static AtomOp newTaggedAtomOp(IExpr address, Register register, IExpr value,
+                                             IOpBin op, String mo, String scope) {
+            AtomOp atom = new AtomOp(address, register, value, op, mo); // mo = ACQ_REL || RLX
+            atom.addFilters(scope);
+            return atom;
+        }
+
+        public static RedOp newTaggedRedOp(IExpr address, Register register, IExpr value,
+                                           IOpBin op, String mo, String scope) {
+            RedOp red = new RedOp(address, register, value, op);
```

Comment:
The `RedOp` event will eventually be transformed (by `Compilation`) to a read and write. Thus this class needs to have a `mo`. Once you add a `mo`, the tag will be automatically added thus not need to add it here.

Reply:
Accepted